core/domain/stats_jobs_one_off.py

```diff
@@ -231,21 +231,21 @@ def _apply_state_name_changes(cls, prev_stats_dict, change_list):
             dict. A dict representation of an ExplorationStatsModel
                 with updated state_stats_mapping and version.
         """
+        exp_versions_diff = exp_domain.ExplorationVersionsDiff(change_list)
+
         # Handling state additions, deletions and renames.
-        for change_dict in change_list:
-            if change_dict['cmd'] == exp_domain.CMD_ADD_STATE:
-
-                prev_stats_dict['state_stats_mapping'][change_dict[
-                    'state_name']] = (stats_domain.StateStats
-                                      .create_default())
-            elif change_dict['cmd'] == exp_domain.CMD_DELETE_STATE:
-                prev_stats_dict['state_stats_mapping'].pop(change_dict[
-                    'state_name'])
-            elif change_dict['cmd'] == exp_domain.CMD_RENAME_STATE:
-                prev_stats_dict['state_stats_mapping'][
-                    change_dict['new_state_name']] = (
-                        prev_stats_dict['state_stats_mapping'].pop(
-                            change_dict['old_state_name']))
+        for state_name in exp_versions_diff.added_state_names:
```

Comment:
I think you've got this in the wrong order. What if a changelist renames A --> B then adds A?

Reply:
Changed order to delete, rename and then add!